src/erc20FeeProxy.ts

```diff
@@ -32,6 +32,16 @@ export function handleTransferWithReferenceAndFee(
 ): void {
   log.info("feeProxy for tx {}", [event.transaction.hash.toHexString()]);
   let payment = createPaymentForFeeProxy(event);
-
   payment.save();
+  const paymentReference = event.params.paymentReference;
+  const escrowId = generateEscrowId(paymentReference);
+  const escrow = Escrow.load(escrowId);
+  if (escrow) {
+    escrow.escrowState = "closed";
+    escrow.payee = payment.to;
+    escrow.save();
+    const escrowClosedEvent = createEscrowEvent(event, paymentReference);
+    escrowClosedEvent.eventType = "closeEscrow";
+    escrowClosedEvent.save();
+  }
```

Comment:
When Escrow is paid out/closed, this event fires from ERC20FeeProxy.
Here we are detecting if an escrow exists with this reference and close it if does.